android/src/org/kiwix/kiwixmobile/ZimFileSelectFragment.java

```diff
@@ -114,25 +114,34 @@ public static void finishResult(String path) {
     }
   }
 
+  @Override
+  public void onResume() {
+    refreshFragment();
+    super.onResume();
+  }
+
   public void refreshFragment(){
     // Of course you will want to faActivity and llLayout in the class and not this method to access them in the rest of
     // the class, just initialize them here
     if (mZimFileList == null)
       return;
 
-    mZimFileList.addFooterView(progressBar);
     mZimFileList.setOnItemClickListener(this);
     mZimFileList.setOnItemLongClickListener(this);
 
     bookDao = new BookDao(KiwixDatabase.getInstance(context));
+    ArrayList<LibraryNetworkEntity.Book> books = bookDao.getBooks();
+    Collections.sort(books, new fileComparator());
 
```

Comment:
We should call mFiles.clear() here, otherwise it may take some time for the list to update if a file is deleted.